### DISCUSSION TITLE
Upgrade guidelines

### DISCUSSION THREAD

**Discussion by Laith-jalal** [pgbouncer/pgbouncer#1014] [Q&A] [answered]
Hi,
I have been checking the usage.md and config.md in order to upgrade my PGbouncer, I got a little bit lost in how to perform the upgrade without dropping the connections.
Do I find a document that explains the upgrading process?

**Comment by JelteF** [comment, marked as answer]
Currently there is no way of performing an in place upgrade without dropping connections, but that is going to be improved with this PR once it's merged: https://github.com/pgbouncer/pgbouncer/pull/902

You can do upgrades if you have multiple nodes with a load balancer in front. And create new nodes with the new version, and add them to the load balancer and then drain the existing nodes before disconnecting them.
